components/dialog/dialog-mixin.js

```diff
@@ -407,7 +407,10 @@ export const DialogMixin = superclass => class extends RtlMixin(superclass) {
 			if (!activeElement
 			|| !isComposedAncestor(dialog, activeElement)
 			|| !activeElement.classList.contains('focus-visible')) {
-				this._focusInitial();
+				// wait till the dialog is visible for Safari
+				requestAnimationFrame(() => {
```

Comment:
The `isVisible` helper checks both `display` and `visibility` properties, however the dialog has `visibility: hidden;` until a frame after `line 402` runs, so we need to delay applying focus until the dialog is actually visible. Alternatively, we could update `isVisible` to not check `visibility`, or take an option to control whether or not it checks it.

Reply:
Nit: could just be `requestAnimationFrame(() => this._focusInitial())`.

Interesting though. I think this makes the most sense, since if we told it to ignore `visibility: hidden` things, then technically the consumer could have hidden their own things and it would ignore them eh?

Reply:
Cool, will update. I need to sort out a potential regression with the mobile filter dropdown anyway.

Yeah, technically we'd want to skip any `visibility: hidden` elements anyway, and the previous `offsetParent` check wouldn't have handled that scenario the way we'd want - it would just silently failed to apply focus, since browsers won't focus them. Having said that, we pretty much never use `visibility` to hide stuff except when managing animations.